src/components/live-tag/live-tag.tsx

```diff
@@ -7,49 +7,111 @@ export enum LiveTagStates {
   Preview = 'Preview',
 }
 
-export interface LiveTagProps {
-  state: LiveTagStates;
+interface LiveTagProps {
+  liveTagState: LiveTagStates;
+}
+
+interface LiveTagState {
+  liveTagState: LiveTagStates;
   isOnLiveEdge: boolean;
-  onClick: () => void;
 }
 
-export class LiveTag extends Component<LiveTagProps> {
-  shouldComponentUpdate(nextProps: LiveTagProps) {
+interface Context {
+  player: any;
+}
+
+export class LiveTag extends Component<LiveTagProps, LiveTagState> {
+  constructor(props: LiveTagProps, context: Context) {
+    super();
+    this.state = {
+      liveTagState: props.liveTagState,
+      isOnLiveEdge: context.player.isOnLiveEdge(),
+    };
+  }
+
+  shouldComponentUpdate(nextProps: LiveTagProps, nextState: LiveTagState) {
     return (
-      this.props.state !== nextProps.state ||
-      this.props.isOnLiveEdge !== nextProps.isOnLiveEdge
+      nextState.liveTagState !== this.state.liveTagState ||
+      nextState.isOnLiveEdge !== this.state.isOnLiveEdge
     );
   }
-  private _getStyles = () => {
-    const {isOnLiveEdge, state} = this.props;
-    if (isOnLiveEdge && state === LiveTagStates.Live) {
-      return styles.live;
+
+  componentWillMount() {
+    this.context.player.addEventListener(
```

Comment:
use eventManager from kalturaPlayer.core and call to removeAll on componentWillUnmount

Reply:
conversation resolved since we don't have "kaltura-player-js" as a dependency